packages/databricks-sdk-js/src/apierr.ts

```diff
@@ -0,0 +1,141 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+const transientErrorStringMatches = [
+    "com.databricks.backend.manager.util.UnknownWorkerEnvironmentException",
+    "does not have any associated worker environments",
+    "There is no worker environment with id",
+    "Unknown worker environment",
+    "ClusterNotReadyException",
+    "connection reset by peer",
+    "TLS handshake timeout",
+    "connection refused",
+    "Unexpected error",
+    "i/o timeout",
+];
+
+export class HttpError extends Error {
+    constructor(readonly message: string, readonly code: number) {
+        super(message);
+    }
+}
+
+export class ApiError extends Error {
+    constructor(
+        public readonly message: string,
+        public readonly errorCode: string,
+        public readonly statusCode: number,
+        public readonly response: any
+    ) {
+        super(message);
+    }
```

Comment:
The stack trace shows a bunch of stuff before showing the actual function where the error is raised.
```
Error: 4465905737085886 unauthorized to perform ReadAction on /org_admin_conf/enableProjectTypeInWorkspace
      at parseErrorFromResponse (src\apierr.ts:5:45)
      at fn (src\api-client.ts:5:2007)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at retry (src\retries\retries.ts:2:5792)
      at ApiClient.request (src\api-client.ts:5:573)
```

One fix is, instead of returning the `ApiError` from `parseError` we instead return `ApiErrorDetails`. We then use the pattern,
`throw new ApiError(parseApiError)`. This will give a much cleaner stack trace.